lib/collection/src/shards/local_shard/mod.rs

```diff
@@ -846,39 +846,53 @@ impl LocalShard {
 
     pub async fn local_shard_info(&self) -> CollectionInfoInternal {
         let collection_config = self.collection_config.read().await.clone();
-        let segments = self.segments().read();
         let mut vectors_count = 0;
         let mut indexed_vectors_count = 0;
         let mut points_count = 0;
         let mut segments_count = 0;
         let mut status = CollectionStatus::Green;
         let mut schema: HashMap<PayloadKeyType, PayloadIndexInfo> = Default::default();
-        for (_idx, segment) in segments.iter() {
-            segments_count += 1;
+        let mut optimizer_status = OptimizersStatus::Ok;
+
+        {
+            let segments = self.segments().read();
+            for (_idx, segment) in segments.iter() {
+                segments_count += 1;
 
-            let segment_info = segment.get().read().info();
+                let segment_info = segment.get().read().info();
 
-            if segment_info.segment_type == SegmentType::Special {
-                status = CollectionStatus::Yellow;
+                if segment_info.segment_type == SegmentType::Special {
+                    status = CollectionStatus::Yellow;
+                }
+                vectors_count += segment_info.num_vectors;
+                indexed_vectors_count += segment_info.num_indexed_vectors;
+                points_count += segment_info.num_points;
+                for (key, val) in segment_info.index_schema {
+                    schema
+                        .entry(key)
+                        .and_modify(|entry| entry.points += val.points)
+                        .or_insert(val);
+                }
             }
-            vectors_count += segment_info.num_vectors;
-            indexed_vectors_count += segment_info.num_indexed_vectors;
-            points_count += segment_info.num_points;
-            for (key, val) in segment_info.index_schema {
-                schema
-                    .entry(key)
-                    .and_modify(|entry| entry.points += val.points)
-                    .or_insert(val);
+            if !segments.failed_operation.is_empty() || segments.optimizer_errors.is_some() {
+                status = CollectionStatus::Red;
+            }
+
+            if let Some(error) = &segments.optimizer_errors {
+                optimizer_status = OptimizersStatus::Error(error.to_string());
             }
-        }
-        if !segments.failed_operation.is_empty() || segments.optimizer_errors.is_some() {
-            status = CollectionStatus::Red;
         }
 
-        let optimizer_status = match &segments.optimizer_errors {
-            None => OptimizersStatus::Ok,
-            Some(error) => OptimizersStatus::Error(error.to_string()),
-        };
+        // If still green while optimization conditions are triggered, mark as grey
```

Comment:
check for status is missing
![Screenshot from 2024-04-03 16-26-04](https://github.com/qdrant/qdrant/assets/1935623/9fec90b0-873b-418c-8493-5d1588b0e5e6)